firmware/src/widgets/stockWidget.cpp

```diff
@@ -102,27 +106,71 @@ void StockWidget::displayStock(int8_t displayIndex, StockDataModel &stock, uint3
     TFT_eSPI &display = m_manager.getDisplay();
 
     display.fillScreen(TFT_BLACK);
-    display.setTextColor(TFT_WHITE);
-    display.setTextSize(4);  // Set text size
+
 
     // Calculate center positions
     int screenWidth = display.width();
     int centre = 120;
+    int arrowOffsetX = 0;
+    int arrowOffsetY = -109;
+    String currencySymbol = "$";
+    if (stock.getCurrency() == "EUR")
+    {
+       currencySymbol = "EUR";
+    } else if (stock.getCurrency() == "GBP")
+    {
+       currencySymbol = "GBP";
+    }
+    
+    
+
+// Outputs
+
+
+
+
+    display.fillRect(0,70, screenWidth, 49, TFT_WHITE);  // rgb565 colors
+    display.fillRect(0,111, screenWidth, 15, TFT_LIGHTGREY);  // rgb565 
+    display.setTextSize(1);  // Set text size
+    display.setTextColor(TFT_WHITE);
+    display.drawString("52 Week:", centre, 188, 1);
+    display.setTextDatum(ML_DATUM);
+    display.drawString("H: " + currencySymbol + stock.getHP(), 89, 201, 1);
```

Comment:
This creates a string (and formats the float) each time the stocks are drawn, is that intentional?

Reply:
Intentional yes. Would this not be needed for it all to be drawn on one single line as needed? If we wanted to draw the strings and floats separately it'd need two call them both separately, making alignment allot harder.